packages/headless/src/features/insight-user-actions/insight-user-actions-preprocessing.test.ts

```diff
@@ -0,0 +1,645 @@
+      const ticketCreationDate = JSON.stringify(caseCreationDate.getTime());
+      const sessionsTimeline = splitActionsIntoTimelineSessions(
+        mappedAndSortedActions,
+        ticketCreationDate
+      );
+
+      const filteredTimeline = filterTimelineActions(
+        sessionsTimeline,
+        actionsToExclude
+      );
+
+      // Checks that no sessions in each section of the timeline contains any of the actions to exclude
+      actionsToExclude.forEach((actionType) => {
+        filteredTimeline.precedingSessions.forEach((session) => {
+          expect(
+            session.actions.every((action) => action.actionType !== actionType)
+          ).toEqual(true);
+          expect(
+            session.actions.every((action) => action.query !== '')
+          ).toEqual(true);
+        });
+
+        filteredTimeline.followingSessions.forEach((session) => {
+          expect(
+            session.actions.every((action) => action.actionType !== actionType)
+          ).toEqual(true);
+          expect(
+            session.actions.every((action) => action.query !== '')
+          ).toEqual(true);
+        });
+
+        filteredTimeline.session?.actions.forEach((action) => {
+          expect(action.actionType !== actionType).toEqual(true);
+          expect(action.query !== '').toEqual(true);
+        });
+      });
+    });
+  });
+
+  describe('#preprocessActionsData', () => {
+    describe('when ticket creation date is not provided', () => {
+      it('should return an empty timeline', async () => {
+        const expectedTimeline = {
+          precedingSessions: [],
+          session: undefined,
+          followingSessions: [],
+        };
+        const mockState = {
+          excludedCustomActions: ['CUSTOM'],
+          ticketCreationDate: undefined,
+          loading: false,
+        };
+
+        const timeline = preprocessActionsData(mockState, fakeActions);
+
+        expect(timeline).toEqual(expectedTimeline);
+      });
+    });
+
+    describe('when the ticket creation date is provided', () => {
+      it('should return a timeline with the case creation session', () => {
+        const ticketCreationDate = JSON.stringify(caseCreationDate.getTime());
+        const mockState = {
+          excludedCustomActions: ['useless_event'],
+          ticketCreationDate: ticketCreationDate,
+          loading: false,
+        };
+        const preprocessedTimeline = preprocessActionsData(
+          mockState,
+          fakeActions
+        );
+
+        expect(preprocessedTimeline).toEqual(expectedTimeline);
+        expect(preprocessedTimeline.precedingSessions.length).toEqual(
+          expectedTimeline.precedingSessions.length
+        );
+        expect(preprocessedTimeline.followingSessions.length).toEqual(
+          expectedTimeline.followingSessions.length
+        );
+        expect(preprocessedTimeline.session?.actions.length).toEqual(
+          expectedTimeline.session?.actions.length
+        );
```

Comment:
this checks are useless as you are already performing a deep equality between `preprocessedTimeline ` and `expectedTimeline ` ?